### ISSUE TITLE
Intended use of database pools

### ISSUE THREAD

**Issue by onwhenrdy** [rstudio/pool#191]
While reading the documentation at [https://rstudio.github.io/pool/articles/advanced-pool.html](https://rstudio.github.io/pool/articles/advanced-pool.html), I noticed that it might give the impression that increasing the number of connections in a database pool could help manage "spikes in traffic" (presumably in shiny or plumber applications).

However, since database pools and their underlying connections cannot be shared across process boundaries (e.g., via promise_futures), I am curious about how the intended use-case for database pools was envisioned, aside from the auto-connect feature. As far as I know, database operations (read/write) will always block. One could argue that different connections to the same database could, in theory, be checked out one after another. However, this seems to be an extremely esoteric usage pattern.

I think this information could benetif other (potential) users as well. Thx.

**Comment by jcheng5** [comment]
You’re right that needing multiple connections is for more esoteric use cases, at least until we get in-process async/non-blocking database queries in R. For today it would really be for transactions, with non-blocking/async steps between database queries. For example, SELECT FOR UPDATE, make an API call, UPDATE, COMMIT.

**Comment by onwhenrdy** [comment]
Thank you for the clarification and feedback.

**Comment by beevabeeva** [comment]
> until we get in-process async/non-blocking database queries in R. 

Hi @jcheng5! 
Getting desperate for this feature. Been reading similar threads across various repos all day.
I cannot understand how people are using Shiny in real production apps with multiple users without it.  
I know there are [hacks using mirai ](https://mirai.r-lib.org/articles/databases.html)and I might well end up doing that, but it means I have to stop using pool...
